server/src/controllers/christmas-tree.es6

```diff
@@ -64,162 +64,216 @@ const translatePermitFromClientToDatabase = input => {
+                result['S:Envelope']['S:Body'][0]['ns2:startOnlineCollectionResponse'][0][
+                  'startOnlineCollectionResponse'
+                ][0];
+              const token = startOnlineCollectionResponse.token[0];
+              permit
+                .update({
+                  paygovToken: token
+                })
+                .then(savedPermit => {
+                  return res.status(200).json({
+                    token: token,
+                    permitId: savedPermit.permitId,
+                    payGovUrl: vcapConstants.payGovClientUrl,
+                    tcsAppID: tcsAppID
+                  });
+                })
+                .catch(error => {
+                  throw new Error(error);
+                });
+            } catch (error) {
+              throw new Error(error);
             }
-        }
-      );
+          }
+        });
+      });
     })
     .catch(error => {
       if (error.name === 'SequelizeValidationError') {
-        return res.status(400).json({ errors: error.errors });
+        return res.status(400).json({
+          errors: error.errors
+        });
       } else {
         return res.status(500).send();
       }
     });
 };
 
-christmasTree.getOneGuidelines = (req, res) => {
+const permitResult = (permit, svgData) => {
+  const result = {
+    permitId: permit.permitId,
+    orgStructureCode: permit.orgStructureCode,
+    firstName: permit.firstName,
+    lastName: permit.lastName,
+    emailAddress: permit.emailAddress,
+    quantity: permit.quantity,
+    totalCost: permit.totalCost,
+    status: permit.status,
+    transactionDate: permit.updatedAt,
+    permitImage: svgData,
+    forest: {
+      forestName: permit.christmasTreesForest.forestName,
+      forestAbbr: permit.christmasTreesForest.forestAbbr
+    }
+  };
+  return result;
+};
 
-  treesDb.christmasTreesForests.findOne({
-    where: {
-      forestAbbr: req.params.id
-    },
-    include: [
-      {
-        model: treesDb.christmasTreesForestSpecies,
-        include: [
-          {
-            model: treesDb.species,
-            include: [
-              {
-                model: treesDb.speciesNotes
-              }
-            ]
-          }
-        ]
+christmasTree.getOnePermit = (req, res) => {
+  treesDb.christmasTreesPermits
+    .findOne({
+      where: {
+        permitId: req.params.id
       },
-      {
-        model: treesDb.christmasTreesForestLocations
-      }
-    ],
-    order: [
-      [ treesDb.christmasTreesForestSpecies, treesDb.species, treesDb.speciesNotes, 'display_order', 'ASC' ],
-      [ treesDb.christmasTreesForestLocations, 'description', 'ASC'],
-      [ treesDb.christmasTreesForestSpecies, 'id', 'ASC']
-    ]
-  })
-    .then(app => {
-      if (app) {
-        res.status(200).json(translateGuidelinesFromDatabaseToClient(app));
+      include: [
+        {
+          model: treesDb.christmasTreesForests
+        }
+      ]
+    })
+    .then(permit => {
```

Comment:
this is a super long string of promises that looks like we might want to break out into a number of named functions

Reply:
I've created a task to do this on #168 to be completed in sprint-6